### PR TITLE
[18.09] Revert selective metadata flushing - breaks multiple uploads/outputs.

### PR DESCRIPTION
**Pull request by jmchilton**

### CODE REVIEW

**Review by mvdbeek — APPROVED** [review]

File: test/api/test_tools.py [hunk 1]
```diff
@@ -802,6 +802,21 @@ def test_run_cat1_with_two_inputs(self, history_id):
...
+    @skip_without_tool("mapper_two")
+    @uses_test_history(require_new=False)
+    def test_bam_state_regression(self, history_id):
+        new_dataset1 = self.dataset_populator.new_dataset(history_id, content='123\n456\n789')
+        inputs = {
+            'input1': dataset_to_param(new_dataset1),
+            'reference': dataset_to_param(new_dataset1),
+        }
+        outputs = self._run_and_get_outputs('mapper_two', history_id, inputs)
+        # Assert we have three outputs with 1, 2, and 3 lines respectively.
```
Comment (nsoranzo) [inline test/api/test_tools.py]: There are 4 outputs, and also the rest of the comment seems wrong.

Comment (jmchilton) [reply test/api/test_tools.py]: Yes - must have been copy-paste problems. Thanks for the catch.

**Review by nsoranzo — APPROVED** [review]

**Comment by mvdbeek** [comment]
Thanks @jmchilton !

**Comment by natefoo** [comment]
@jmchilton Should we create an issue to track restoring the functionality? If what you told me yesterday was correct, we'd get a nice performance improvement from this and it's a SQLAlchemy bug that should hopefully be fixed at some point?

**Pull request merged**